client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/HubConnection.spec.ts

```diff
@@ -390,6 +390,31 @@ describe("HubConnection", () => {
             connection.receive({ type: 3, invocationId: connection.lastInvocationId });
             expect(await observer.completed).toEqual([1, 2, 3]);
         });
+
+        it("does not require error function registered", async () => {
```

Comment:
how would these tests fail before the change?

Reply:
```
1) HubConnection stream does not require error function registered
  Message:
    Failed
  Stack:
    Error: Failed
        at assertion.then.catch (E:\ASPNETCORE\signalr\client-ts\dist\test\Microsoft.AspNetCore.SignalR.Client.TS.Tests\
JasmineUtils.js:28:34)
        at runMicrotasksCallback (internal/process/next_tick.js:58:5)
        at _combinedTickCallback (internal/process/next_tick.js:67:7)
        at process._tickCallback (internal/process/next_tick.js:98:9)
  Message:
    Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  Stack:
    Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
        at ontimeout (timers.js:365:14)
        at tryOnTimeout (timers.js:237:5)
        at Timer.listOnTimeout (timers.js:207:5)
```

Reply:
Oh, they are run via the asyncit wrapper... 

can you make the tests more explicit by taking the `done` function and invoking it in the `hubConnection.onclose` handler after making sure that the error is null?

Reply:
I will try!

Reply:
Is there a way to access the normal `it` function when you have `import { asyncit as it, captureException } from './JasmineUtils';`

Reply:
I think you can just do:
```
it("does not require error function registered", async (done) => {
   ...   
  hubConnection.onclose(e => {
      expect(e).toBe(null);
      done();
  })
```

Reply:
Nope, because normal `it` function is hidden by `asyncit as it`

Reply:
I [see](https://github.com/aspnet/SignalR/blob/dev/client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/JasmineUtils.ts#L8) what you are saying

Reply:
Slightly unrelated
https://github.com/aspnet/SignalR/blob/6d9dd3c1cc5fc85667383bf3ce7e4aeda1158162/client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/JasmineUtils.ts#L10
calling `done()` after `fail()` completes failures without waiting for timeout

Reply:
You would have to probably return a promise that is fulfilled in onclose and rejected on an exception. In this case the test would not be async (interesetingly it is marked as async but it does not have any async calls).

Reply:
The idea was to complete the test from onclose handler (as per my code sample). Otherwise we could keep what you originally had since it is basically the same thing as this one...

Reply:
Oh I see. So that doesn't make sense for this test because `e` is always going to have a value since we call `onclose` with an error in order to trigger the `observer.error` call.

Reply:
If this is the case let's get back to the original version. I thought we could end up with a different error than the one we used to close the underlying connection but it does not seem to be the case.
(Sorry for asking for changes that were not needed).

Reply:
I think its at the original version now?

Reply:
Yes, it's good to go. Again, sorry for adding more work.